posts/_posts/2019-08-26-fabric-gotchas.html

```diff
@@ -65,3 +65,11 @@ <h3>Object does not update after changing property - objectCaching</h3>
 </pre>
 <p> For more information and detailed explanation, check the <a href="/fabric-object-caching">dedicated caching page</a>.</p>
 </div>
+
+<h3>Object are not rendered at the given coordinates</h3>
+
+By default, objects have a transparent stroke of width 1 that shift them by 1 pixel horizontally and vertically. To remove the stroke:
+
+<pre>
+rect.set('strokeWidth', 1);
+</pre>
```

Comment:
a couple of notes
- we want probably to set the strokeWidth to 0 and explain that if they do that on the prototype automatically for all objects, SVG import will be broken.

- the offset is strokeWidth / 2, so is 0.5 pixel.

- we want to specify that Objects are slightly off place, `not rendered at given coordinates` can be any offset and could mislead the reader.

THANKS!

Reply:
- Yes, it is strokeWidth 0. I made a mistake. I don't know what is this thing of SVG import.
- I don't get why the offset is strokeWidth/2. In my case, it was an offset of 1.
- Ok

I have modified the PR

Reply:
The offset is half the strokewidth because the stroke is painted over the immaginary non material line of the shape.

So if your rect is drawn at position 0, on the line between the pixel -1 and pixel 0,a line wide 1 pixels is made, this line is half pixel on pixel -1 and half pixel on the pixel 0.

To you it seems an offset of 1 pixel but what you got is one pixel of NON RED color that was also NON BLACK color ( as your background black ). It was a mixed color of the 2, representing a blending at 50%.
If you would set a strokeWidth of 2, leaving the stroke transparent, you would have seen a full black single pixel before the red rect, just one.